@amifun/getArgs.m

```diff
@@ -78,11 +78,11 @@
         case 'y'
             this.argstr = '(realtype t, int it, realtype *y, N_Vector x, void *user_data)';
         case 'z'
-            this.argstr = '(realtype t, int ie, int *nroots, realtype *z, N_Vector x, void *user_data)';
+            this.argstr = '(realtype t, int ie, realtype *z, N_Vector x, void *user_data, void *temp_data)';
```

Comment:
Regarding passing individual pointers or only the Data classes, we currently anyways use all the udata members and do not pass them as arguments and this cannot change due to restrictions from CVODES, consequently we will always have a dependence on UserData which limits reusability and I would thus prefer to only pass UserData/TempData/ReturnData/ExpData as this will simplify the code a bit and make changes easier.